src/views/users/UserRegistrationView.vue

```diff
@@ -1,3 +1,243 @@
+        </div>
+        <div class="input-container orcid-container row" v-if="state.orcid">
+            <div class="col-sm-2 align-self-center">
+                <span>ORCID</span>
+            </div>
+            <div class="col-sm-1">
+                <img alt="ORCID logo" src="/ORCIDiD_iconvector.svg" title="ORCID iD" class="orcid-icon"/>
+            </div>
+            <div class="col-sm-5 align-self-center">
+                <span>{{ state.orcid }}</span>
+            </div>
+        </div>
+        <div class="input-container checkbox-container">
+          <input class="checkbox" v-model="state.accepted_terms_of_use" id="termsOfUseCheckbox" type="checkbox" name="accepted_terms_of_use" value="1" />
+          <b>I have read and accepted the <router-link to="/terms">Morphobank Terms of Use &amp; Privacy Policy</router-link></b>
+        </div>
+        <div v-if="state.errorMessage" class="alert alert-danger">
+          {{ state.errorMessage }}
+        </div>
+        <div>
+          <button class="w-100 btn btn-lg btn-primary form-group" type="submit">
+            Register
+          </button>
+        </div>
+      </div>
+      <div class="input-container space-container"></div>
+      <div class="registration-instructions">
+        <h5>
+            Registration is not required to view published projects. If you are interested in only viewing a project, please<br><br>
+            <router-link to="/project/pub_date">click to BROWSE published projects</router-link><br><br><em>OR</em><br><br>
+            SEARCH published projects in the search box at the top of the page
+        </h5>
+      </div>
+    </form>
+  </div>
 </template>
+
+
+<script setup>
+import { reactive, onMounted } from 'vue'
+import router from '../../router'
+import { useAuthStore } from '@/stores/AuthStore.js'
+import axios from 'axios'
+
+const authStore = useAuthStore()
+
+const state = reactive({
+  fname: '',
+  lname: '',
+  email: '',
+  password: '',
+  password2: '',
+  accepted_terms_of_use: false,
+  orcid: '',
+  accessToken: '',
+  refreshToken: '',
+  errorMessage: null,  // Added new property for error message
+})
+
+onMounted(() => {
+  if (authStore.orcid.name) {
+    let names = authStore.orcid.name.split(' ')
+    state.fname = names[0]
+    state.lname = names.slice(1).join(' ')
+    state.orcid = authStore.orcid.orcid
+    state.accessToken = authStore.orcid.accessToken
+    state.refreshToken = authStore.orcid.refreshToken
+  }
+})
+
+const validateInputFields = () => {
+  if (!state.fname || !state.lname || !state.email || !state.password || !state.password2 || !state.accepted_terms_of_use) {
+    state.errorMessage = 'Please fill all required fields.';
+    return false;
+  }
+
+  const emailValidation = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+  if (!emailValidation.test(state.email)) {
+    state.errorMessage = 'Please enter a valid email address.';
+    return false;
+  }
+
+  return true;
+}
+
+const validatePasswords = () => {
```

Comment:
Can you try making this more dynamic? i.e. Once the user finishes entering the confirm password, we can validate the password and display the error